lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -94,7 +94,17 @@ protected function get_block_classes( $style_nodes ) {
 			}
 
 			// 2. Generate the rules that use the general selector.
-			$block_rules .= static::to_ruleset( $selector, $declarations );
+			// If the block is not used on this page, return.
+			$general_selector_rules = static::to_ruleset( $selector, $declarations );
+			if ( 'styles' === $metadata['path'][0] && 'blocks' === $metadata['path'][1] ) {
+				$block_name = str_replace( 'core/', '', $metadata['path'][2] );
+				wp_add_inline_style(
```

Comment:
It terms of approach, I think we want to tap into the existing API: it already has logic to decide whether to enqueue the block's CSS as a single stylesheet with them all or a separate one per block. Otherwise, we'll need to replicate that logic for the global styles.

In terms of implementation, `WP_Theme_JSON` has remained agnostic about WordPress APIs (`get_stylesheet`, `wp_add_inline_style`, etc), which was the `WP_Theme_JSON_Resolver` responsibility. I think it's benefitial to separate those concerns.

Reply:
> it already has logic to decide whether to enqueue the block's CSS as a single stylesheet with them all or a separate one per block

Do you know where this is? I can't find it!

Reply:
I don't! Looking for `should_load_separate_block_assets` can perhaps help to start the investigation.

Reply:
We could probably use the `wp_enqueue_block_style()` function 🤔